src/Features/CSharp/Portable/BraceCompletion/AbstractCurlyBraceOrBracketCompletionService.cs

```diff
@@ -144,7 +144,7 @@ private static bool ContainsOnlyWhitespace(SourceText text, int openingPosition,
             Debug.Assert(desiredCaretLine.GetFirstNonWhitespacePosition() == null, "the line between the formatted braces is not empty");
 
             // Set the caret position to the properly indented column in the desired line.
-            var newDocument = document.WithChangedText(formattedText, cancellationToken);
+            var newDocument = document.WithChangedRoot(formattedRoot, cancellationToken);
```

Comment:
slightly concerned that i don't really know what this implication of this is.  i'm not pushign back on it.  just worried this may cause something unexpected to happen.

Reply:
"something unexpected" would mean an existing bug getting exposed, right?